### PR TITLE
Java 5189

### PR DESCRIPTION
**Pull request by rozza**
JAVA-5189

### CODE REVIEW

**Comment by rozza** [comment]
Patch: https://spruce.mongodb.com/version/651ebeadd1fe0742372e82b1/

I'm not 100% sure this is the correct approach

File: driver-core/src/main/com/mongodb/internal/Locks.java [hunk 1]
```diff
@@ -52,6 +52,7 @@ public static void lockInterruptibly(final Lock lock) throws MongoInterruptedExc
         try {
             lock.lockInterruptibly();
         } catch (InterruptedException e) {
+            lock.unlock();
```
Comment (jyemin) [inline driver-core/src/main/com/mongodb/internal/Locks.java:55]: If `InterruptedException` is thrown from `lockInterruptibly`, how is it not already unlocked (unless it's been locked by a different thread, in which case that thread should be responsible for unlocking)?

Comment (stIncMale) [reply driver-core/src/main/com/mongodb/internal/Locks.java:55]: This `lock.unlock()` will cause `IllegalMonitorStateException` if `ReentrantLock`/`ReentrantReadWriteLock` is used. If `InterruptedException` is thrown from `Lock.lockInterruptibly`, then the lock hasn't been acquired by the current thread.

**Comment by rozza** [comment]
Closing as it incidently worked rather than fixed the issue. #1220 implements a correct fix.

**Pull request closed without merging**